File: dotwrp.f90
!g95 -fno-underscoring dotwrp.f90 -c
double complex function zdotc_(n, zx, incx, zy, incy)
double complex zx(*), zy(*), z
integer n, incx, incy

call cblas_zdotc_sub(%val(n), zx, %val(incx), zy, %val(incy), z)

zdotc_ = z
return
end

double complex function zdotu_(n, zx, incx, zy, incy)
double complex zx(*), zy(*), z
integer n, incx, incy

call cblas_zdotu_sub(%val(n), zx, %val(incx), zy, %val(incy), z)

zdotu_ = z
return
end

complex function cdotc_(n, cx, incx, cy, incy)
complex cx(*), cy(*), c
integer n, incx, incy

call cblas_cdotc_sub(%val(n), cx, %val(incx), cy, %val(incy), c)

cdotc_ = c
return
end

complex function cdotu_(n, cx, incx, cy, incy)
complex cx(*), cy(*), c
integer n, incx, incy

call cblas_cdotu_sub(%val(n), cx, %val(incx), cy, %val(incy), c)

cdotu_ = c
return
end

real function sdot_(n, sx, incx, sy, incy)
real sx(*), sy(*), s
double precision d
integer n, incx, incy

sdot_ = cblas_sdot(%val(n), sx, %val(incx), sy, %val(incy))

return
end

File: dotwrp.c
// compile with: clang -c dotwrp.c
#include <complex.h>
#include <vecLib/cblas.h>

#ifdef DOTWRP_GLOBAL
#define DYLIB_LOCAL
#else
#define DYLIB_LOCAL __attribute__ ((visibility ("hidden"))) 
#endif

DYLIB_LOCAL
double complex zdotc_( const int* n, const double complex* x, const int* ix, const double complex *y, const int* iy )
{
	double complex z;
	cblas_zdotc_sub( *n, x, *ix, y, *iy, &z );
	return z;
}

DYLIB_LOCAL
double complex zdotu_( const int* n, const double complex* x, const int* ix, const double complex *y, const int* iy )
{
	double complex z;
	cblas_zdotu_sub( *n, x, *ix, y, *iy, &z );
	return z;
}

DYLIB_LOCAL
float complex cdotc_( const int* n, const float complex* x, const int* ix, const float complex *y, const int* iy )
{
	float complex z;
	cblas_cdotc_sub( *n, x, *ix, y, *iy, &z );
	return z;
}

DYLIB_LOCAL
float complex cdotu_( const int* n, const float complex* x, const int* ix, const float complex *y, const int* iy )
{
	float complex z;
	cblas_cdotu_sub( *n, x, *ix, y, *iy, &z );
	return z;
}

DYLIB_LOCAL
float sdot_( const int* n, const float* x, const int* ix, const float *y, const int* iy )
{
	return cblas_sdot( *n, x, *ix, y, *iy );
}

DYLIB_LOCAL
float sdsdot_( const int* n, const float* a, const float* x, const int* ix, const float *y, const int* iy )
{
	return cblas_sdsdot( *n, *a, x, *ix, y, *iy );
}

DYLIB_LOCAL
float snrm2_( const int* n, const float* x, const int* ix )
{
	return cblas_snrm2( *n, x, *ix );
}

DYLIB_LOCAL
float sasum_( const int* n, const float* x, const int* ix )
{
	return cblas_snrm2( *n, x, *ix );
}

DYLIB_LOCAL
float scnrm2_( const int* n, const float complex* x, const int* ix )
{
	return cblas_scnrm2( *n, x, *ix );
}

DYLIB_LOCAL
float scasum_( const int* n, const float complex* x, const int* ix )
{
	return cblas_scasum( *n, x, *ix );
}


